addons/sale_project/models/project.py

```diff
@@ -399,16 +401,88 @@ def _get_revenues_items_from_sol(self, domain=None, with_action=True):
             'total': {'to_invoice': total_to_invoice, 'invoiced': total_invoiced},
         }
 
+    def _get_revenues_items_from_invoices(self, excluded_move_line_ids=None):
+        """
+        Get all revenues items from invoices, and put them into their own
+        "other_invoice_revenues" section.
+        If the final total is 0 for either to_invoice or invoiced (ex: invoice -> credit note),
+        we don't output a new section
+
+        :param excluded_move_line_ids a list of 'account.move.line' to ignore
+        when fetching the move lines, for example a list of invoices that were
+        generated from a sales order
+        """
+        if excluded_move_line_ids is None:
+            excluded_move_line_ids = []
+        query = self.env['account.move.line'].sudo()._search([
+            ('analytic_account_id', 'in', self.analytic_account_id.ids),
+            ('move_id.move_type', 'in', ['out_invoice', 'out_refund']),
+            ('parent_state', 'in', ['draft', 'posted']),
+            ('price_subtotal', '>', 0),
+            ('id', 'not in', excluded_move_line_ids)
+        ])
+        # account_move_line__move_id is the alias of the joined table account_move in the query
+        # we can use it, because of the "move_id.move_type" clause in the domain of the query, which generates the join
+        # this is faster than a search_read followed by a browse on the move_id to retrieve the move_type of each account.move.line
+        query_string, query_param = query.select('price_subtotal', 'parent_state', 'account_move_line__move_id.move_type')
+        self._cr.execute(query_string, query_param)
+        invoices_move_line_read = [moves for moves in self._cr.dictfetchall()]
```

Comment:
The result of `self._cr.dictfetchall()` is not already an array? 🤔

Reply:
ditto